### PR TITLE
Add support for removing query params in request mutations

### LINKED ISSUE #294: Add support for redacting query params in request mutations
The [current implementation of Request Mutation](https://github.com/google/fhir-gateway/blob/main/server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java#L283-L293) only supports adding extra parameters to the request.

The proposal is to also support removing existing parameters that need not be forwarded to the proxied upstream server e.g. HAPI FHIR Server. 

One benefit of this is to make sure we don't flood the proxied server with parameters only meant for the proxy(gateway)

### PR DESCRIPTION
**Pull request by ndegwamartin**
## Description of what I changed

Closes #294 which Adds support for removing query params in request mutations

## E2E test

TESTED:

Please replace this with a description of how you tested your PR beyond the
automated e2e/unit tests.

## Checklist: I completed these to help reviewers :)

- [x] I have read and will follow the [review process](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/blob/master/doc/review_process.md).
- [x] I am familiar with Google Style Guides for the language I have coded in.

  No? Please take some time and review [Java](https://google.github.io/styleguide/javaguide.html) and [Python](https://google.github.io/styleguide/pyguide.html) style guides.

- [x] My IDE is configured to follow the Google [**code styles**](https://google.github.io/styleguide/).

  No? Unsure? -> [configure your IDE](https://github.com/google/google-java-format).

- [x] I have **added tests** to cover my changes. (If you refactored existing code that was well tested you do not have to add tests)
- [x] I ran `mvn clean package` right before creating this pull request and added all formatting changes to my commit.
- [x] All new and existing **tests passed**.
- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Review by bashir2 — APPROVED** [review]
Thanks @ndegwamartin for adding this feature.

File: server/src/main/java/com/google/fhir/gateway/interfaces/RequestMutation.java [hunk 1]
```diff
@@ -32,4 +33,8 @@ public class RequestMutation {
   // parameter in RequestMutation is already present in the original request).
   // Old parameter values should be explicitly retained while mutating values for that parameter.
   @Builder.Default Map<String, List<String>> queryParams = new HashMap<>();
```
Comment (bashir2) [inline server/src/main/java/com/google/fhir/gateway/interfaces/RequestMutation.java:35]: nit: Consider changing this to `extraQueryParams` or `additionalQueryParams` to make its purpose more clear (not a blocker as it is not directly added in this PR).

Comment (ndegwamartin) [reply server/src/main/java/com/google/fhir/gateway/interfaces/RequestMutation.java]: Done

**Comment by ndegwamartin** [comment]
@bashir2 bumped up the artifact version to 0.4.0 instead of 0.3.3 because the [Rename RequestMutation queryParams variable](https://github.com/google/fhir-gateway/pull/308/commits/88db45f060ac6c9b843e50603559dc6660c5bb8b) needs some slight refactor for compatibility on dependent apps. Let me know if we should use the latter instead.

**Pull request merged**